src/EnergyPlus/ChillerAbsorption.cc

```diff
@@ -1541,24 +1541,35 @@ namespace ChillerAbsorption {
 		}
 
 		if ( GeneratorInletNode > 0 ) {
-			//         Hot water plant is used for the generator
 			if ( BLASTAbsorber( ChillNum ).GenHeatSourceType == NodeType_Water ) {
-				CpFluid = GetSpecificHeatGlycol( PlantLoop( BLASTAbsorber( ChillNum ).GenLoopNum ).FluidName, Node( GeneratorInletNode ).Temp, PlantLoop( BLASTAbsorber( ChillNum ).GenLoopNum ).FluidIndex, RoutineName );
-				if ( ( BLASTAbsorber( ChillNum ).FlowMode == ConstantFlow ) || ( BLASTAbsorber( ChillNum ).FlowMode == NotModulated ) ) {
-					SteamMassFlowRate = BLASTAbsorber( ChillNum ).GenMassFlowRateMax;
-				} else {
-					SteamMassFlowRate = QGenerator / CpFluid / BLASTAbsorber( ChillNum ).GeneratorDeltaTemp;
+				Real64 GenMassFlowRate = 0.0;
+				int GenLoopNum = BLASTAbsorber( ChillNum ).GenLoopNum;
+				int GenLoopSideNum = BLASTAbsorber( ChillNum ).GenLoopSideNum;
+				//  Hot water plant is used for the generator
+				CpFluid = GetSpecificHeatGlycol( PlantLoop( GenLoopNum ).FluidName, Node( GeneratorInletNode ).Temp, PlantLoop( GenLoopSideNum ).FluidIndex, RoutineName );
+				if ( PlantLoop( GenLoopNum ).LoopSide( GenLoopSideNum ).FlowLock == 0 ) {
+					if ( (BLASTAbsorber( ChillNum ).FlowMode == ConstantFlow) || (BLASTAbsorber( ChillNum ).FlowMode == NotModulated) ) {
+						GenMassFlowRate = BLASTAbsorber( ChillNum ).GenMassFlowRateMax;
+					} else { // LeavingSetpointModulated
+						// since the .FlowMode applies to the chiller evaporator, the generater mass flow rate will be proportional to the evaporator mass flow rate
+						Real64 GenFlowRatio = EvapMassFlowRate / BLASTAbsorber( ChillNum ).EvapMassFlowRateMax;
+						GenMassFlowRate = min( BLASTAbsorber( ChillNum ).GenMassFlowRateMax, GenFlowRatio * BLASTAbsorber( ChillNum ).GenMassFlowRateMax );
+					}
+				} else { // If FlowLock is True
+					GenMassFlowRate = Node( GeneratorInletNode ).MassFlowRate;
```

Comment:
The FlowLock logic isn't needed in the component model.  The component model should just calculate a flow to request and call SetComponentFlowRate.  If FlowLock is false on this loop, that method will set this request on the loop structure, and return a suitable flow based on the original request and any flow limits.  If FlowLock is true on this loop then SetComponentFlowRate will just return the inlet node flow rate.  If there are any other changes I find on this review, then this will have to be remedied and will push this PR past the freeze.  If there isn't any other changes to be made on this branch, then it can go in.